benchmark/FindCXXStandard.cmake

```diff
@@ -16,38 +16,32 @@ file(WRITE "${SOURCE_FILE}"
     "int main() { }                                                 \n"
 )
 
-# Check for C++14
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+message(STATUS "Check for C++14")
 try_compile(_supports_14 ${CMAKE_BINARY_DIR} ${SOURCE_FILE}
     CMAKE_FLAGS -DCMAKE_CXX_STANDARD=14
     COMPILE_DEFINITIONS -DSTANDARD_DATE=201402L
 )
 if (_supports_14)
     set(CXX_STANDARD 2014)
+    set(CMAKE_CXX_STANDARD 14)
+    message(STATUS "Check for C++14 -- works")
     return()
 endif()
 
-# Check for C++11
+message(STATUS "Check for C++11")
 try_compile(_supports_11 ${CMAKE_BINARY_DIR} ${SOURCE_FILE}
     CMAKE_FLAGS -DCMAKE_CXX_STANDARD=11
     COMPILE_DEFINITIONS -DSTANDARD_DATE=201103L
 )
 if (_supports_11)
     set(CXX_STANDARD 2011)
+    set(CMAKE_CXX_STANDARD 11)
+    message(STATUS "Check for C++11 -- works")
     return()
 endif()
 
-# Check for C++98
-try_compile(_supports_98 ${CMAKE_BINARY_DIR} ${SOURCE_FILE}
-    CMAKE_FLAGS -DCMAKE_CXX_STANDARD=98
-    COMPILE_DEFINITIONS -DSTANDARD_DATE=199711L
-)
-if (_supports_98)
-    set(CXX_STANDARD 1998)
-    return()
-endif()
-
-# Emit an error if none of the standards are satisfied.
-message(FATAL_ERROR
-    "FindCXXStandard.cmake: The C++ compiler '${CMAKE_CXX_COMPILER}' appears "
-    "not to support any of the known standard (1998, 2011, 2014)."
-)
+message(STATUS "Fall back to C++98")
+set(CXX_STANDARD 1998)
+set(CMAKE_CXX_STANDARD 98)
```

Comment:
Testing for C++98 support wouldn't hurt, if it weren't for buggy compilers such as GCC 4.6 which sets `__cplusplus` to `1L`, but since we set `CMAKE_CXX_STANDARD_REQUIRED`, it's better we let CMake error out, should someone try running the benchmark suite for 20 years old compilers which indeed don't support C++98.